mg_vtable.h

```diff
@@ -105,42 +105,42 @@ enum {		/* pass one of these to get_vtbl */
 
 #ifdef DOINIT
 EXTCONST char * const PL_magic_vtable_names[magic_vtable_max] = {
-    "arylen",
-    "arylen_p",
-    "backref",
-    "checkcall",
-    "collxfrm",
-    "dbline",
-    "debugvar",
-    "defelem",
-    "destruct",
-    "env",
-    "envelem",
-    "hints",
-    "hintselem",
-    "hook",
-    "hookelem",
-    "isa",
-    "isaelem",
-    "lvref",
-    "mglob",
-    "nkeys",
-    "nonelem",
-    "ovrld",
-    "pack",
-    "packelem",
-    "pos",
-    "regdata",
-    "regdatum",
-    "regexp",
-    "sig",
-    "sigelem",
-    "substr",
-    "sv",
-    "taint",
-    "utf8",
-    "uvar",
-    "vec"
+    [want_vtbl_arylen] = "arylen",
+    [want_vtbl_arylen_p] = "arylen_p",
+    [want_vtbl_backref] = "backref",
+    [want_vtbl_checkcall] = "checkcall",
+    [want_vtbl_collxfrm] = "collxfrm",
+    [want_vtbl_dbline] = "dbline",
+    [want_vtbl_debugvar] = "debugvar",
+    [want_vtbl_defelem] = "defelem",
+    [want_vtbl_destruct] = "destruct",
+    [want_vtbl_env] = "env",
+    [want_vtbl_envelem] = "envelem",
```

Comment:
Array designated initializers are not C++ compatible, and since we have people who build perl with a C++ compiler, it's excluded by the guide in `perlhacktips.pod`.

g++ supports it as an extension. but MSVC doesn't.

Reply:
@leonerd, @tonycoz identified some problems with this p.r. back in March.  Do you expect to move forward with the p.r.?  (It has begun to acquire merge conflicts.)  Thanks.